azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebApp.java

```diff
@@ -0,0 +1,302 @@
+import com.microsoft.azure.toolkits.appservice.service.AbstractAppServiceUpdater;
+import com.microsoft.azure.toolkits.appservice.service.IAppServicePlan;
+import com.microsoft.azure.toolkits.appservice.service.IWebApp;
+import com.microsoft.azure.toolkits.appservice.service.IWebAppDeploymentSlot;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class WebApp implements IWebApp {
+
+    private static final String UNSUPPORTED_OPERATING_SYSTEM = "Unsupported operating system %s";
+    private WebAppEntity entity;
+    private AzureAppService azureAppService;
+
+    private AzureResourceManager azureClient;
+    private com.azure.resourcemanager.appservice.models.WebApp webAppClient;
+
+    public WebApp(WebAppEntity entity, AzureAppService azureAppService) {
+        this.entity = entity;
+        this.azureAppService = azureAppService;
+        this.azureClient = azureAppService.getAzureResourceManager();
+    }
+
+    @Override
+    public WebAppEntity entity() {
+        return entity;
+    }
+
+    @Override
+    public IAppServicePlan plan() {
+        return azureAppService.appServicePlan(getWebAppClient().appServicePlanId());
+    }
+
+    @Override
+    public WebAppCreator create() {
+        return new WebAppCreator();
+    }
+
+    @Override
+    public void start() {
+        getWebAppClient().start();
+    }
+
+    @Override
+    public void stop() {
+        getWebAppClient().stop();
+    }
+
+    @Override
+    public void restart() {
+        getWebAppClient().restart();
+    }
+
+    @Override
+    public void delete() {
+        azureClient.webApps().deleteById(getWebAppClient().id());
+    }
+
+    @Override
+    public void deploy(File file) {
+        deploy(AppServiceUtils.getDeployTypeByFileExtension(file), file);
+    }
+
+    public void deploy(DeployType deployType, File target) {
+        getWebAppClient().deploy(com.azure.resourcemanager.appservice.models.DeployType.fromString(deployType.getValue()), target);
+    }
+
+    @Override
+    public boolean exists() {
+        return getWebAppClient(true) != null;
+    }
+
+    @Override
+    public String hostName() {
+        return getWebAppClient().defaultHostname();
+    }
+
+    @Override
+    public String state() {
+        return getWebAppClient().state();
+    }
+
+    @Override
+    public PublishingProfile getPublishingProfile() {
+        final com.azure.resourcemanager.appservice.models.PublishingProfile publishingProfile = getWebAppClient().getPublishingProfile();
+        return PublishingProfile.createFromServiceModel(publishingProfile);
+    }
+
+    @Override
+    public Runtime getRuntime() {
+        return AppServiceUtils.getRuntimeFromWebApp(getWebAppClient());
+    }
+
+    @Override
+    public WebAppUpdater update() {
+        return new WebAppUpdater();
```

Comment:
it is un-completed implementation ?

Reply:
No, return Updater in `update()` method, and commit by the commit method of Updater